In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.dummy import DummyClassifier
from sklearn.metrics import accuracy_score


df = pd.read_csv('data.csv')


X = df.drop(columns=['diagnosis', 'id', 'Unnamed: 32'], errors='ignore')
y = df['diagnosis']


X_train, X_test, y_train, y_test = train_test_split(
    X, y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

baseline_model = DummyClassifier(strategy='most_frequent')
baseline_model.fit(X_train, y_train)


y_pred = baseline_model.predict(X_test)


accuracy = accuracy_score(y_test, y_pred)

print(f"Liczba cech (kolumn w X): {X.shape[1]}")
print(f"Baseline Accuracy: {accuracy:.4f}")

Liczba cech (kolumn w X): 30
Baseline Accuracy: 0.6316


Accuracy Baseline
W tym zbiorze najczęstszą klasą jest B (Benign – łagodny). Baseline mówi nam: "Jaki wynik osiągnę, jeśli nie będę niczego analizował i każdemu pacjentowi z góry powiem, że ma nowotwór łagodny?".

Wynik ten wyniesie prawdopodobnie około 0.6316 (czyli 63%).